### ISSUE TITLE
深度学习框架

### ISSUE THREAD

**Issue by Kinpzz** [Kinpzz/Deep-Learning-on-Medical-Image#7]
# 深度学习框架选择
* Caffe
* Caffe2
* Tensorflow( wrapper : Keras)
* torch( in lua)
* pytorch

之前接触过Tensorflow和caffe，参考了大家的一点看法，目前2017年pytorch可能是比较好的一个框架。
之前学习keras觉得很简洁很好写，但是存在不足的是包装过度不适合做research，存在效率问题。而Tensorflow虽然有着强大的社区支持，但是对图的定义、debug自己在写的过程中也感觉到了复杂。

在接下来做research的过程中，要继续深入学习一下caffe，以及过度到caffe2（拥有更好的效率和速度）。以及接触学习facebook在2017年年初刚发布的pytorch。

**Comment by Kinpzz** [comment]
# Caffe
## 依赖包
### ProtoBuffer
ProtoBUffer是由Google开发的一种可以实现内存与非易失存储介质（如硬盘文件）交换的协议接口。Caffe源码中大量使用ProtoBuffer作为权值和模型参数的载体。
#参考
* 21天实战caffe

**Comment by Kinpzz** [comment]
# GPU 加速
GPU:（Graphics Processing Unit）
## CUDA
CUDA（Compute Unified Device Architecture）是NVIDIA在2006年推出的一套针对异构计算资源（Intel CPU + NVIDIA GPU）下的大规模并行计算的架构，包括编译器（nvcc）、开发工具、运行时库和驱动等模块。
CUDA语法和C语言高度像素，大大降低了GPU编程的门槛。使用的时候直接利用CUDA提供的cuBLAS软件库进行运算即可。

## cuDNN
NVIDIA从2014年开始推出了面向深度学习的专用加速库cuDNN，支持常见的深度学习计算模型（卷积、下采样、非线性、Softmax）
